### PR TITLE
Adding Pythia 8

### PR DESCRIPTION
**Pull request by henryiii**
This adds a Pythia8 recipe. Currently, this is mostly the default options, but with Python activated. MacOS and Linux should be supported (Pythia8 doesn't support Windows as far as I know). No dependencies have been activated with ROOT in either direction (both packages have options to add support for the other).

@chrisburr, @bluehood, @egpbos, let me know if you want to be added as maintainers. Also, if you know any Pythia8 people that might be interested, feel free to ping them.

For review, I assume that would be @conda-forge/help-python-c. I've tested this locally on macOS.

### CODE REVIEW

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -1,83 +0,0 @@
-# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
-
-# Jinja variables help maintain the recipe as you'll update the version only here.
-# Using the name variable with the URL in line 13 is convenient
-# when copying and pasting from another recipe, but not really needed.
-{% set name = "simplejson" %}
```
Comment (chrisburr) [inline recipes/example/meta.yaml]: The example recipe shouldn't be removed.

File: recipes/pythia8/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - pythia8
+
+about:
+  home: http://home.thep.lu.se/Pythia/
+  license: GPL-2.0
+  license_family: GPL
+  license_file: COPYING
+  summary: 'A tool for the generation of events in high-energy collisions'
+
+  # The remaining entries in this section are optional, but recommended
```
Comment (chrisburr) [inline recipes/pythia8/meta.yaml]: This comment can be removed

File: recipes/pythia8/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+  run:
+    - python
+
+test:
+  # Some package might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
```
Comment (chrisburr) [inline recipes/pythia8/meta.yaml]: These comments can be removed

File: recipes/pythia8/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      the evolution from a few-body hard process to a complex multiparticle
+      final state. It contains a library of hard processes, models for initial-
+      and final-state parton showers, matching and merging methods between hard
+      processes and parton showers, multiparton interactions, beam remnants,
+      string fragmentation and particle decays. It also has a set of utilities
+      and several interfaces to external programs.
+
+extra:
+  recipe-maintainers:
+    - henryiii
```
Comment (chrisburr) [inline recipes/pythia8/meta.yaml]: Yeah, I'm happy to be a maintainer.

```suggestion
    - henryiii
    - chrisburr
```

**Comment by henryiii** [comment]
> Are there any environment variables that should be added in a post setup hook?

I don't think any are required for the standalone version; but I have seen `PYTHIA8` and `PYTHIA8DATA` used by ROOT.  I think I'll add those two.

**Pull request merged**